client/my-sites/patterns/hooks/use-patterns.ts

```diff
@@ -66,7 +67,42 @@ export function getPatternsQueryOptions(
 export function usePatterns(
 	locale: string,
 	category: string,
-	queryOptions: Omit< UseQueryOptions< Pattern[] >, 'queryKey' > = {}
+	queryOptions: Omit< UseQueryOptions< Pattern[] >, 'queryKey' | 'select' > = {}
 ) {
-	return useQuery< Pattern[] >( getPatternsQueryOptions( locale, category, queryOptions ) );
+	const queryClient = useQueryClient();
+	const queryResult = useQuery< Pattern[] >(
+		getPatternsQueryOptions( locale, category, queryOptions )
+	);
+
+	// When users submit a search, we fetch *all* patterns. This callback groups the returned
+	// patterns by category and populates the react-query cache for each category.
+	useEffect( () => {
+		if ( category === '' && queryResult.data?.length ) {
```

Comment:
`category === ''` means that we run it every time when a users visits home page. WDYT about removing `category` from dependencies of the hood and introduce something like `isFirstVisitOfHomePage`?

Reply:
Yeah, it means we run it every time the home page is rendered (if `queryResult.dataUpdatedAt` differs from the last render). I don't think that's really a problem, though. We only prepopulate the category queries if they don't already have cached data.